chef/cookbooks/bind9/templates/default/named.conf.erb

```diff
@@ -49,6 +54,6 @@ include "/etc/bind/named.conf.default-zones";
 include "/etc/bind/named.conf.crowbar";
 include "/etc/bind/named.conf.local";
 include "/etc/named.d/rndc-access.conf";
-<% if @enable_designate -%>
+<% if @enable_designate and @dns_master -%>
```

Comment:
is this an intentional change?

Reply:
Yeah, this was intentional. If you go look at the changed line 407 of 'chef/cookbooks/bind9/recipes/default.rb' you'll see that confusingly `@enable_designate` wan't simply the value of `node[:dns][:enable_designate]`. It's:

`enable_designate: node[:dns][:enable_designate] && node[:dns][:master]`

In the old version of the template this is OK. Because when it's the DNS master and enable_designate we want to incloud the rndc key. But that's not the case anymore. We now want to listen on the public ip if designate is enabled **and** if designate is enabled and the node also happens to be the DNS_master then we want to drop the rndc.

So I separated the `&&` and passed them in separately so we could do the right thing. But the and on line 57.